lib/mist/s3/buckets.py

```diff
@@ -37,21 +38,41 @@ def handle_delete(self):
         try:
             patterns = [re.compile(arg) for arg in args[1:]]
             buckets = self.find_matching(patterns)
-            bucket_names = [{"bucket": bucket.name} for bucket in buckets]
             try:
-                while len(buckets) > 0:
-                    bucket = buckets[0]
+                for bucket in buckets:
                     bucket.delete()
-                    buckets = buckets[1:]
             except Exception as e:
-                Logger.error("Error deleting S3 bucket '%s': %s" & (buckets[0].name, e))
+                Logger.error("Error deleting S3 bucket '%s': %s" % (buckets[0].name, e))
```

Comment:
Is `buckets[0]` the actual bucket that you want to display in this error? Do you have access to the `bucket` that caused the exception in this scope?